core/src/serve_repair.rs

```diff
@@ -62,9 +90,9 @@ pub struct ServeRepairStats {
 /// Window protocol messages
 #[derive(Serialize, Deserialize, Debug)]
 pub enum RepairProtocol {
-    WindowIndex(ContactInfo, u64, u64),
-    HighestWindowIndex(ContactInfo, u64, u64),
-    Orphan(ContactInfo, u64),
+    WindowIndex(ContactInfo, u64, u64, Nonce),
```

Comment:
Fortunately RepairProtocol is already an enum, so we can leave the existing three and add another three:
```
    WindowIndexWithNonce(ContactInfo, u64, u64, Nonce),
    HighestWindowIndexWithNonce(ContactInfo, u64, u64, Nonce),
    OrphanWithNonce(ContactInfo, u64, Nonce),
```
Then the code can support both variants for a time, enabling a rolling update

Reply:
Oh yeah that would be a lot less work to code too, whew, thanks!